File: examples/test_hashlib.py
# Test hash-like functions (manual implementations without hashlib)
from typing import List


def simple_hash(data: str) -> int:
    """Compute a simple hash of a string (djb2-like algorithm)."""
    h: int = 5381
    i: int = 0
    while i < len(data):
        ch: int = ord(data[i])
        h = ((h * 33) + ch) % 1000000007
        i = i + 1
    return h


def hash_to_hex_digits(value: int) -> str:
    """Convert an integer hash value to a hex-like string of digits."""
    if value < 0:
        value = 0 - value
    result: str = ""
    remaining: int = value
    i: int = 0
    while i < 16:
        digit: int = remaining % 16
        if digit < 10:
            result = str(digit) + result
        else:
            result = str(digit) + result
        remaining = remaining // 16
        i = i + 1
    return result


def hash_password(password: str) -> int:
    """Hash a password using simple hash."""
    return simple_hash(password)


def verify_integrity(data: str, expected_hash: int) -> int:
    """Verify data integrity. Returns 1 if match, 0 if not."""
    actual: int = simple_hash(data)
    if actual == expected_hash:
        return 1
    return 0


def hash_combine(h1: int, h2: int) -> int:
    """Combine two hash values."""
    return ((h1 * 31) + h2) % 1000000007


def test_module() -> int:
    """Run all tests."""
    passed: int = 0

    # Test simple_hash produces consistent values
    h1: int = simple_hash("hello")
    h2: int = simple_hash("hello")
    if h1 == h2:
        passed = passed + 1

    # Test different strings produce different hashes
    h3: int = simple_hash("world")
    if h1 != h3:
        passed = passed + 1

    # Test hash_password
    hp: int = hash_password("secret")
    if hp > 0:
        passed = passed + 1

    # Test verify_integrity
    test_data: str = "test data"
    test_hash: int = simple_hash(test_data)
    if verify_integrity(test_data, test_hash) == 1:
        passed = passed + 1
    if verify_integrity("wrong data", test_hash) == 0:
        passed = passed + 1

    # Test hash_combine
    combined: int = hash_combine(h1, h3)
    if combined > 0:
        passed = passed + 1

    # Test empty string
    h_empty: int = simple_hash("")
    if h_empty == 5381:
        passed = passed + 1

    return passed

File: examples/test_hashlib.rs
#[doc = "// Python import: hashlib"] #[doc = "Hash a password using SHA256"] #[doc = " Depyler: verified panic-free"] #[doc = " Depyler: proven to terminate"] pub fn hash_password<'a>(password: & 'a str)  -> String {
    let hasher = std::Sha256();
    for(k, v) in password.encode("utf-8".to_string()) {
    hasher.insert(k, v)
};
    return hasher.hexdigest();
   
}
#[doc = "Compute MD5 checksum of file data"] #[doc = " Depyler: verified panic-free"] #[doc = " Depyler: proven to terminate"] pub fn compute_file_checksum<'a>(data: & 'a bytes)  -> String {
    let hasher = std::Md5();
    for(k, v) in data {
    hasher.insert(k, v)
};
    return hasher.hexdigest();
   
}
#[doc = "Verify data integrity using SHA512"] #[doc = " Depyler: verified panic-free"] #[doc = " Depyler: proven to terminate"] pub fn verify_integrity<'a, 'b>(data: & 'a str, expected_hash: & 'b str)  -> bool {
    let hasher = std::Sha512();
    for(k, v) in data.encode("utf-8".to_string()) {
    hasher.insert(k, v)
};
    let actual_hash = hasher.hexdigest();
    let _cse_temp_0 = actual_hash == expected_hash;
    return _cse_temp_0
}